Sources/TensorFlow/Core/TensorHandle.swift

```diff
@@ -212,3 +212,94 @@ public struct VariantHandle {
         self.handle = handle
     }
 }
+
+//===------------------------------------------------------------------------------------------===//
+// TensorBuffer based on a C `TF_Tensor*`.
+//===------------------------------------------------------------------------------------------===//
+
+// TF Tensor-specific initializer.
+internal class CTensorTensorBuffer<Scalar> : TensorBuffer<Scalar> {
+    let cTensor: CTensor
+
+    /// Creates a local tensor buffer from a C `TF_Tensor*` value and takes ownership of the value.
+    init(owning cTensor: CTensor, count: Int) {
+        debugLog("Initializing TensorBuffer with a cTensor of \(count) elements.")
+        let actualCount = (0..<TF_NumDims(cTensor)).reduce(1) { accumulator, next in
+            accumulator * Int(TF_Dim(cTensor, next))
+        }
+        assert(actualCount == count)
```

Comment:
Can't we get rid of the `count` argument from the `init` as it is implied from `cTensor`?

Reply:
I believe the existing code wants to ensure that the count matches the shape as a consistency check.